diff --git a/lib/message/converter.js b/lib/message/converter.js
index 27a659a3290fadd990a1a980918515a6ded4978f..d6156587a02b1ed441ac8d2a3e2104e6343306b2 100644
--- a/lib/message/converter.js
+++ b/lib/message/converter.js
@@ -111,7 +111,7 @@ class Converter {
             return;
         }
         if (qq === "all") {
-            var q = 0, flag = 1, display = "全体成员";
+            var q = 0, flag = 1, display = text || "全体成员";
         }
         else {
             var q = Number(qq), flag = 0, display = text || String(qq);
@@ -121,7 +121,6 @@ class Converter {
                 display = member?.card || member?.nickname || display;
             }
         }
-        display = "@" + display;
         if (dummy)
             return this._text(display);
         const buf = Buffer.allocUnsafe(6);
