diff --git a/utils/image-load-url.js b/utils/image-load-url.js
--- a/utils/image-load-url.js
+++ b/utils/image-load-url.js
@@ -1,6 +1,11 @@
 const https = require('https')
+const fs = require('fs')
 
 module.exports = (url, filter = false) => {
+  if (typeof url === 'string' && url.startsWith('/')) {
+    return Promise.resolve(fs.readFileSync(url))
+  }
+
   return new Promise((resolve, reject) => {
     const options = new URL(url)
     options.headers = {
